### PR TITLE
encoder: Add utility processors

### PR DESCRIPTION
**Pull request by kingthorin**
Added processors for: lower case, remove whitespace, reverse, upper case.

- CHANGELOG > Add note.
- encoder.html > Added details.
- Encoders & UnitTests > Added new functionality/coverage.

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

**Comment by kingthorin** [comment]
Addressed conflict and upper/lower case terminology.

File: addOns/encoder/src/test/java/org/zaproxy/addon/encoder/processors/predefined/utility/LowerCaseUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+class LowerCaseUnitTest extends ProcessorTests<LowerCase> {
+
+    @Override
+    protected LowerCase createProcessor() {
+        return LowerCase.getSingleton();
+    }
+
+    @Test
+    void shouldEncodeSimpleScriptTag() throws Exception {
```
Comment (thc202) [inline addOns/encoder/src/test/java/org/zaproxy/addon/encoder/processors/predefined/utility/LowerCaseUnitTest.java:38]: Merge them? Like done with the other tests.

File: addOns/encoder/src/test/java/org/zaproxy/addon/encoder/processors/predefined/utility/UppercaseUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+class UppercaseUnitTest extends ProcessorTests<UpperCase> {
+
+    @Override
+    protected UpperCase createProcessor() {
+        return UpperCase.getSingleton();
+    }
+
+    @Test
+    void shouldEncodeSimpleScriptTag() throws Exception {
```
Comment (thc202) [inline addOns/encoder/src/test/java/org/zaproxy/addon/encoder/processors/predefined/utility/UppercaseUnitTest.java:38]: Same here.

**Comment by kingthorin** [comment]
Tweaked.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Pull request merged**